### PR TITLE
Allow reading otel context from reactor ContextView

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9263
Also removes latest dep limit from reactor tests.
This PR presents an alternative to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11156 that introduces separate library instrumentation for reactor 3.4.

### CODE REVIEW

**Comment by laurit** [comment]
cc @123liuziming

File: instrumentation/reactor/reactor-3.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/v3_4/ContextPropagationOperator34InstrumentationTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                  reactorContext, Context.current());
+          Context otelContext =
+              ContextPropagationOperator.getOpenTelemetryContextFromContextView(
+                  newReactorContext, null);
+          assertThat(otelContext).isNotNull();
+          Span.fromContext(otelContext).setAttribute("foo", "bar");
+          Context otelContext2 =
+              ContextPropagationOperator.getOpenTelemetryContext(newReactorContext, null);
+          assertThat(otelContext2).isNotNull();
+          Span.fromContext(otelContext2).setAttribute("foo", "bar");
```
Comment (zeitlinger) [inline instrumentation/reactor/reactor-3.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/v3_4/ContextPropagationOperator34InstrumentationTest.java:42]: why are we setting `foo=bar` on both contexts?

Comment (laurit) [reply instrumentation/reactor/reactor-3.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/v3_4/ContextPropagationOperator34InstrumentationTest.java]: change it to use separate attributes

**Review by zeitlinger — APPROVED** [review]

File: instrumentation/reactor/reactor-3.1/library/src/test/java/io/opentelemetry/instrumentation/reactor/v3_1/ReactorCoreTest.java [hunk 2]
```diff
@@ -160,13 +160,16 @@ void fluxInNonBlockingPublisherAssembly() {
...
     int result =
         testing.runWithSpan(
             "parent",
             () ->
                 Mono.defer(
                         () -> {
-                          Span.current().setAttribute("middle", "foo");
+                          if (!testLatestDeps) {
+                            Span.current().setAttribute("middle", "foo");
+                          }
```
Comment (trask) [inline instrumentation/reactor/reactor-3.1/library/src/test/java/io/opentelemetry/instrumentation/reactor/v3_1/ReactorCoreTest.java:172]: Can you add comment why we don’t test this against latest?

File: instrumentation/reactor/reactor-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/v3_1/BaseMonoWithSpanTest.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ void nested() {
                     span ->
                         span.hasName("inner-manual")
                             .hasKind(SpanKind.INTERNAL)
-                            .hasParent(trace.getSpan(1))
+                            .hasParent(trace.getSpan(Boolean.getBoolean("testLatestDeps") ? 0 : 1))
```
[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/reactor/reactor-3.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/reactor/v3_1/BaseMonoWithSpanTest.java:91]: I think this is a change in how the framework operates, in latest dep version the test does not behave as initially expected.

**Pull request merged**